btr-web/btr-main-app/components/individual-person/tax-info/TaxNumber.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <div class="flex flex-col py-5">
+    <div class="flex items-center mb-2 py-1">
+      <URadio
+        id="taxNumberRadioButton"
+        v-model="selectedButton"
+        :value="`CRA Tax Number`"
+        @change="handleRadioButtonChange(`CRA Tax Number`)"
+      />
+      <UFormGroup :name="name" class="ml-5">
+        <UInput
+          v-model="taxNumber"
+          type="text"
+          :disabled="selectedButton !== 'CRA Tax Number'"
+          variant="bcGov"
+          :placeholder="$t('placeholders.taxNumber')"
+          class="w-80"
+          @blur="formatInput"
+        />
+      </UFormGroup>
+    </div>
+    <div class="flex items-center mb-2 py-2">
+      <URadio
+        id="noTaxNumberRadioButton"
+        v-model="selectedButton"
+        :value="`No Tax Number`"
+        @change="handleRadioButtonChange(`No Tax Number`)"
+      />
+      <label class="ml-5"> This person does not have a CRA Tax Number </label>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, defineProps, defineEmits, watch } from 'vue'
+
+defineProps({
+  id: { type: String, required: true },
+  name: { type: String, default: 'name' },
+  modelValue: {
+    type: Object,
+    default: () => ({
+      hasTaxNumber: false,
+      taxNumber: ''
+    })
+  },
+  label: { type: String, default: '' }
+})
+
+const emit = defineEmits<{(e: 'update:modelValue', value: { hasTaxNumber: boolean; taxNumber: string }): void}> ()
+
+// Indicate which radio button is selected.
+const selectedButton = ref('')
+
+// The tax number input value
+const taxNumber = ref('')
+
+// Watch the selected radio button value to update the taxNumber value and emit the parent's `v-model`.
+watch(() => selectedButton.value, (button) => {
+  if (button === 'No Tax Number') {
+    emit('update:modelValue', { hasTaxNumber: false, taxNumber: 'No Tax Number' })
```

Comment:
tax number should be either '' or null here. Or even better, you set `taxNumber.value` to '' then just use it in emit also. That makes sure that it is propagated to what is set.